feat(slider)!: migrating tokens

bernhard-adobe opened the pull request:
Core tokens migration for Slider.
Includes t-shirt sizing

## Description
Migrated and modded all vars.
Added t-shirt sizing 
 
ticket: https://jira.corp.adobe.com/projects/CSS/issues/CSS-177

## Issues Addressed
- [Slider Handle Animation Issue 2917](https://github.com/adobe/spectrum-web-components/issues/2917)
- [Slider Cursor Pointer Issue 2918](https://github.com/adobe/spectrum-web-components/issues/2918)
- [Slider Controls Margin Issue 711](https://github.com/adobe/spectrum-css/issues/711)
- [Slider Tick Alignment Issue 2990](https://github.com/adobe/spectrum-web-components/issues/2990)


## How and where has this been tested?
 - **How this was tested:** comparing http://localhost:3000/docs/slider.html vs https://opensource.adobe.com/spectrum-css/slider.html#withlabel
 - **Browser(s) and OS(s) this was tested with:** Chrome Version 110.0.5419.3 (Official Build) canary (arm64)


## Screenshots
<img width="402" alt="Screen Shot 2022-11-15 at 22 36 00" src="https://user-images.githubusercontent.com/52184321/202092975-c89cdde1-9d34-4701-8dcb-4d5d3de9f155.png">
<img width="874" alt="Screen Shot 2022-11-15 at 22 36 02" src="https://user-images.githubusercontent.com/52184321/202092986-6b5ca63b-f800-430c-b85f-0f8f288ab003.png">

Firefox:
<img width="1812" alt="Screen Shot 2022-11-15 at 22 39 31" src="https://user-images.githubusercontent.com/52184321/202093131-859fb28e-83c5-412c-bfe0-bc2f22d33623.png">

Safari:
<img width="1628" alt="Screen Shot 2022-11-15 at 22 39 33" src="https://user-images.githubusercontent.com/52184321/202093139-04377b8f-5e28-43c5-845a-3d3384a63136.png">


## QA notes

- [ ] Validate circle alignment on tick marks
- [ ] Validate circle alignment at edges


## To-do list
- [x] If my change impacts other components, I have tested to make sure they don't break.
- [x] If my change impacts documentation, I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING document](/.github/CONTRIBUTING.md).
- [x] I have tested these changes in Windows High Contrast mode.
- [ ] This pull request is ready to merge.

lunarfusion reviewed (changes requested):
Ignore anything that isn't helpful, of course!

bernhard-adobe commented on components/slider/index.css:
```diff
@@ -133,9 +165,8 @@ governing permissions and limitations under the License.
-      border-end-start-radius: var(--spectrum-slider-track-border-radius);
-      border-start-end-radius: 0;
+      border-start-start-radius: var(--mod-slider-track-corner-radius, var(--spectrum-slider-track-corner-radius));
+      border-end-start-radius: var(--mod-slider-track-corner-radius, var(--spectrum-slider-track-corner-radius))) border-start-end-radius: 0;
```
typo. good catch.

bernhard-adobe commented on components/slider/index.css:
```diff
@@ -366,52 +403,48 @@ governing permissions and limitations under the License.
-    1px
-  );
+  margin: 0 var(--mod-slider-track-margin-offset, var(--spectrum-slider-track-margin-offset));
+  margin-block-start: calc(var(--mod-slider-tick-mark-height, var(--spectrum-slider-tick-mark-height)) -2px);
```
I tried your variant and that gives me these mis-aligned slider ticks as the `  align-items: center;` is applied to the whole slider. 
![Screen Shot 2022-11-28 at 23 05 36](https://user-images.githubusercontent.com/52184321/204436253-e4b99c2b-95d7-4426-b501-e50b3b38ad31.png)

bernhard-adobe commented on components/slider/index.css:
```diff
@@ -144,38 +175,43 @@ governing permissions and limitations under the License.
-      border-start-end-radius: var(--spectrum-slider-track-border-radius);
-      border-end-end-radius: var(--spectrum-slider-track-border-radius);
+      border-start-start-radius: var(--mod-slider-track-corner-radius, var(--spectrum-slider-track-corner-radius));
+      border-end-start-radius: var(--mod-slider-track-corner-radius, var(--spectrum-slider-track-corner-radius));
```
thanks! I fixed it. Good eye!
works for both standard and range.
![Screen Shot 2022-11-29 at 19 08 04](https://user-images.githubusercontent.com/52184321/204678225-e9b3006d-d17e-48aa-b342-c57bf2520c15.png)
![Screen Shot 2022-11-29 at 19 08 48](https://user-images.githubusercontent.com/52184321/204678230-604c17ee-044d-4cd8-bbe2-0c46a8e24a8e.png)

bernhard-adobe commented:
@lunarfusion I addressed your feedback! Thanks a lot for the impressive review, excellent work!

The only thing missing here is the top centering that I can't get to work as mentioned above. The handle aligns in the center of the entire Slider height, not just the center of the track. 
Did you try this out in the browser or did you check out the branch locally and pasted in the code?

Thanks for pointing me to a good solution!

pfulton reviewed (changes requested):
It looks like there might be a mismatch between the track color and the tick color (zoomed in to see the difference) - or - maybe there's a `z-index` issue here?

<img width="501" alt="Screen Shot 2022-12-01 at 4 13 56 PM" src="https://user-images.githubusercontent.com/360251/205160486-5dcad909-b2d7-4558-968d-db0db0d551d7.png">

Also, with regard to the tick position & the handle, there's [an open PR](https://github.com/adobe/spectrum-css/pull/1289) from last year that you might be able to get some info from.

castastrophe commented on components/slider/index.css:
```diff
@@ -184,43 +214,34 @@ governing permissions and limitations under the License.
         border-start-end-radius: 0;
         border-end-end-radius: 0;
+
+        /* border-start-start-radius: var(--mod-slider-track-corner-radius, var(--spectrum-slider-track-corner-radius));
```
Is this content still needed or can it be cleaned up before merge?

bernhard-adobe replied:
lets remove it.

castastrophe commented on components/slider/index.css:
```diff
@@ -244,36 +265,43 @@ governing permissions and limitations under the License.
   }
+
+  &+.spectrum-Slider-handle {
+    margin-block: calc(var(--mod-slider-bottom-to-handle, var(--spectrum-slider-bottom-to-handle)) * -1 -5px + var(--mod-slider-margin-top-offset, var(--spectrum-slider-margin-top-offset))) 0;
```
```suggestion
    margin-block: calc(
    var(--mod-slider-bottom-to-handle, var(--spectrum-slider-bottom-to-handle)) * -1 - 5px + var(--mod-slider-margin-top-offset, var(--spectrum-slider-margin-top-offset))) 0;
```

I noticed the missing space between the subtraction & 5px was causing the calc to render without it.

bernhard-adobe replied:
good catch! thanks for seeing this.

bernhard-adobe commented on components/slider/index.css:
```diff
@@ -10,63 +10,95 @@ OF ANY KIND, either express or implied. See the License for the specific languag
+
+.spectrum-Slider--sizeS {
+
+  --spectrum-slider-handle-size: var(--mod-slider-handle-size-small, var(--spectrum-slider-handle-size-small));
```
@pfulton let me know if we still need those -mods for items that already have the -mod options from previous assignments.

pfulton replied:
Wow, I completely missed this question, and I apologize for not seeing it and addressing it until now.

Overall, I'd like us to avoid declaring `--mod` custom property overrides in initial declarations and to override "global" tokens.

Avoid this:
```css
.spectrum-Slider {
  --spectrum-slider-handle-border-radius: var(--mod-corner-radius-200, var(--spectrum-corner-radius-200));}
```

Instead, do something like this:
```css
.spectrum-Slider {
  --spectrum-slider-handle-border-radius: var(--spectrum-corner-radius-200);}

/* later, when it's actually used: */
  border-radius: var( --mod-slider-handle-border-radius, var(--spectrum-slider-handle-border-radius));
```

The implications are a bit nuanced here, but due to the cascade, it should be a little safer to do something like I did in my suggestion above. A `--mod-corner-radius-200` could potentially be more wide-reaching than `--mod-slider-handle-border-radius`.

Plus, IMO, this gets things a bit more inline with the "API" we're providing to consumers for a given component (they're "modifying" Slider related custom properties rather than a "global" Spectrum corner radius).

That's a big explanation, but I think we should change the approach here.

bernhard-adobe commented on components/slider/index.css:
```diff
@@ -244,36 +265,43 @@ governing permissions and limitations under the License.
   position: absolute;
   inset-inline-start: 0;
-  inset-block-start: calc(var(--spectrum-slider-height) / 2);
+  inset-block-start: calc(var(--mod-slider-height-size, var(--spectrum-slider-height-size)) / 2);
+  inset-block-start: calc(var(--mod-component-bottom-to-text-75, var(--spectrum-component-bottom-to-text-75)) + var(--mod-slider-control-height, var(--spectrum-slider-control-height)) / 2);
```
oopps, removed

bernhard-adobe commented on components/slider/index.css:
```diff
@@ -10,63 +10,93 @@ OF ANY KIND, either express or implied. See the License for the specific languag
+  --spectrum-slider-input-top-size: calc(var(--mod-slider-handle-size, var(--spectrum-slider-handle-size)) / -2 / 4);
+
+
+  /* aka 80px */
```
let's remove it, was a fixed value that was missing from tokens.

bernhard-adobe commented:
> It looks like there might be a mismatch between the track color and the tick color (zoomed in to see the difference) - or - maybe there's a `z-index` issue here?
> 
> <img alt="Screen Shot 2022-12-01 at 4 13 56 PM" width="501" src="https://user-images.githubusercontent.com/360251/205160486-5dcad909-b2d7-4558-968d-db0db0d551d7.png">
> 
> Also, with regard to the tick position & the handle, there's [an open PR](https://github.com/adobe/spectrum-css/pull/1289) from last year that you might be able to get some info from.

Nah, it's just the wrong color. we don't use tokens for the ticks and even in the current version of slider there is a color bug for the disabled ticks.
<img width="479" alt="Screen Shot 2022-12-05 at 20 26 26" src="https://user-images.githubusercontent.com/52184321/205805677-c8c860df-9aa4-453d-85c6-861100e732da.png">



I have fixed the wrong color. good catch.
<img width="394" alt="Screen Shot 2022-12-05 at 19 51 21" src="https://user-images.githubusercontent.com/52184321/205797303-69535840-c9c1-4f28-885d-036f1b2ee998.png">
<img width="394" alt="Screen Shot 2022-12-05 at 19 51 21" src="https://user-images.githubusercontent.com/52184321/205797329-2cc8d888-7269-410f-a134-f808245707d5.png">

bernhard-adobe commented on components/slider/index.css:
```diff
@@ -244,36 +265,43 @@ governing permissions and limitations under the License.
 
 .spectrum-Slider-handle {
   position: absolute;
   inset-inline-start: 0;
```
actually it does, it sets the ramp handle at the right spot. 
Here some coloring with a border and a background to illustrate:
<img width="582" alt="Screen Shot 2022-12-05 at 20 29 50" src="https://user-images.githubusercontent.com/52184321/205806187-501b92d8-ced5-4419-b3c9-88f10298de01.png">

bernhard-adobe commented on components/slider/index.css:
```diff
@@ -244,36 +265,43 @@ governing permissions and limitations under the License.
 
-  border-width: var(--spectrum-slider-handle-border-size);
+  margin-block: calc(var(--mod-slider-bottom-to-handle, var(--spectrum-slider-bottom-to-handle)) * -1 + var(--mod-slider-margin-top-offset, var(--spectrum-slider-margin-top-offset))) 0;
+  margin-inline: calc(var(--mod-slider-handle-size, var(--spectrum-slider-handle-size)) / -2) 0;
```
Awesome, thank you  @castastrophe ! 
<img width="455" alt="Screen Shot 2022-12-05 at 20 57 08" src="https://user-images.githubusercontent.com/52184321/205809618-daa58843-036b-441f-858c-0361072eeb18.png">

pfulton replied:
I'm not sure that this suggestion actually got implemented (or, at least it doesn't seem to be in the codebase, so maybe it didn't get committed?) because I still see the misalignment. Mind taking a quick look?
<img width="279" alt="Screen Shot 2022-12-08 at 11 10 47 AM" src="https://user-images.githubusercontent.com/360251/206502019-a6c1b41e-2fa0-4426-ba6b-8a57d3314f0d.png">

bernhard-adobe replied:
I used
` margin-inline: calc(var(--mod-slider-handle-size, var(--spectrum-slider-handle-size)) / -2) 0;`

thank you @castastrophe again.

pfulton reviewed (changes requested):
Thanks for fixing up the positioning of the tick and the handle!
---
I noticed a few new things:

It looks like the spacing on the right side of the handle has been adversely affected by the recent changes:

<img width="298" alt="Screen Shot 2022-12-06 at 10 15 54 AM" src="https://user-images.githubusercontent.com/360251/205951243-f291773a-2304-44dc-b651-78da6cc356c3.png">

It's particularly noticeable on the slider with two handles:
<img width="299" alt="Screen Shot 2022-12-06 at 10 16 05 AM" src="https://user-images.githubusercontent.com/360251/205951520-d717b419-357f-4a4a-988f-a77bcd2f4377.png">

Also, the tick/track color difference seems to still be an issue:
<img width="277" alt="Screen Shot 2022-12-06 at 10 21 06 AM" src="https://user-images.githubusercontent.com/360251/205951937-a8b406a9-0589-4c07-b3e0-d356da260ade.png">

Finally, would you mind double-checking on the Windows High Contrast Mode styles? It looks like (at least) the standard, non-disabled version of the track color needs to be fixed:

Your branch:
<img width="1115" alt="Screen Shot 2022-12-06 at 10 22 34 AM" src="https://user-images.githubusercontent.com/360251/205952652-59e56178-4e22-4a0d-8572-7a658ce782ba.png">

Current production:
<img width="714" alt="Screen Shot 2022-12-06 at 10 23 04 AM" src="https://user-images.githubusercontent.com/360251/205952748-7653ba06-d2da-4bee-ba90-d3b88be2706f.png">

pfulton reviewed (changes requested):
We're almost there on the Windows High Contrast Mode styles! Two things I noticed: the ticks are missing, and the ramp seems to be different now vs. what's live in production.

Your branch:
<img width="356" alt="Screen Shot 2022-12-08 at 11 10 28 AM" src="https://user-images.githubusercontent.com/360251/206502614-3cb7626a-437d-4166-ad22-38a424a75acb.png">

Production:
<img width="355" alt="Screen Shot 2022-12-08 at 11 10 16 AM" src="https://user-images.githubusercontent.com/360251/206502677-2c661bf5-c6f9-4ede-b5a2-83d0582ff1f3.png">

Could you fix up those examples?

bernhard-adobe commented:
`I'm not sure that this suggestion actually got implemented (or, at least it doesn't seem to be in the codebase, so maybe it didn't get committed?) because I still see the misalignment. Mind taking a quick look?`

@pfulton this actually adds a wider space between the slider handle on the right side between the handle and the track. I would suggest to maybe not implement this solution.
<img width="877" alt="Screen Shot 2022-12-08 at 12 56 25" src="https://user-images.githubusercontent.com/52184321/206555792-27a6152d-e0a2-4dba-a58c-2c3acf48943c.png">

pfulton commented:
> @pfulton this actually adds a wider space between the slider handle on the right side between the handle and the track. I would suggest to maybe not implement this solution.

Ok. Because this has been a long-standing issue with this component in our docs site, we still need to fix this problem before we merge the PR in. If you get stuck on finding a solution, please mention it in the `spectrum_css_dev` channel so that more folks can lend a hand.

castastrophe commented:
In case they haven't been highlighted yet, I wanted to make sure we'd addressed the following PRs here since their updates won't apply once this merges: #1005, #1133, #1206, #1289

bernhard-adobe commented:
>  Thanks @castastrophe .
We got the following PRs covered in https://jira.corp.adobe.com/browse/CSS-332 Clean up old Slider PRs:
 https://github.com/adobe/spectrum-css/pull/1206 and https://github.com/adobe/spectrum-css/pull/1289

I will add to the ticket the PRs and ping @pfulton about the updates.
https://github.com/adobe/spectrum-css/issues/1005, https://github.com/adobe/spectrum-css/issues/1133

pfulton commented on components/slider/index.css:
```diff
@@ -455,17 +482,269 @@ governing permissions and limitations under the License.
+  top: calc(var(--mod-slider-control-height, var(--spectrum-slider-control-height)) / 2 - 1px);
 
-  margin-left: calc(var(--spectrum-slider-handle-width) / 2 * -1);
+  margin-left: calc(var(--spectrum-slider-handle-size) / 2 * -1);
```
```suggestion
  margin-inline-start: calc(var(--spectrum-slider-handle-size) / 2 * -1);
```

pfulton replied:
☝🏻 as this is, when switching to RTL, the track is incorrectly positioned. I think switching to the logical property will resolve that.

<img width="314" alt="Screen Shot 2023-01-18 at 4 28 52 PM" src="https://user-images.githubusercontent.com/360251/213299063-071a1184-63c9-48e0-87e2-5b55106d32c7.png">

bernhard-adobe replied:
Spot on @pfulton .
I am getting to the point where paddings and margins are a rare exception to the inline-start for RLT.   
<img width="363" alt="Screen Shot 2023-01-23 at 13 34 33" src="https://user-images.githubusercontent.com/52184321/214144192-653e3220-f1e7-457c-bc86-1974fbc7575a.png">

pfulton commented on components/slider/index.css:
```diff
@@ -455,17 +482,269 @@ governing permissions and limitations under the License.
 .spectrum-Slider-handleContainer,
 .spectrum-Slider-trackContainer {
-  width: calc(100% + var(--spectrum-slider-handle-width));
+  width: calc(100% + var(--spectrum-slider-handle-size));
```
```suggestion
  inline-size: calc(100% + var(--spectrum-slider-handle-size));
```
Let's switch to the logical property for a bit of future-proofing.

pfulton commented on components/slider/index.css:
```diff
@@ -455,17 +482,269 @@ governing permissions and limitations under the License.
 
 .spectrum-Slider-trackContainer {
-  height: var(--spectrum-slider-height);
+  height: var(--mod-slider-height-size, var(--spectrum-slider-height-size));
```
```suggestion
  block-size: var(--mod-slider-height-size, var(--spectrum-slider-height-size));
```
Let's use the logical property here for some future-proofing.

pfulton reviewed (changes requested):
A handful of (hopefully!) quick logical property requests here, @bernhard-adobe - but this is looking awesome otherwise.

I didn't call them all out, but would you mind taking a look for other places where you're using `height` and `width` and switch those to logical properties? (I think you were good on `margin` and `padding` references)

Thanks!

pfulton commented:
@bernhard-adobe would you mind also taking a peek at this new issue (#1583) related to Slider?

bernhard-adobe commented:
> @bernhard-adobe would you mind also taking a peek at this new issue (#1583) related to Slider?

of course. I added the downstate state in https://github.com/adobe/spectrum-css/pull/1547/commits/77917cdb319903d670d3162efd938977b9046c0e. It was indeed missing and should be added now.
![2023-01-20_10-23-40 (1)](https://user-images.githubusercontent.com/52184321/213769844-66bfe43f-e73f-4246-bdef-a839a1ddf1c3.gif)

pfulton commented on components/slider/index.css:
```diff
@@ -10,63 +10,94 @@ OF ANY KIND, either express or implied. See the License for the specific languag
+
+
+/* missing tokens for variant with Tick, also need t-shirt sizing */
+.spectrum--medium {
+  --spectrum-slider-tick-mark-height: 10px;
+  /* --spectrum-global-dimension-size-125 */
+  --spectrum-slider-tick-mark-width: 2px;
+  /* --spectrum-global-dimension-size-25: */
+  --spectrum-slider-tick-mark-border-radius: 2px;
+  /* --spectrum-global-dimension-size-25 2px for Spectrum,   --spectrum-global-dimension-size-10 1px for Express */
+  --spectrum-slider-ramp-track-height: 16px;
+  /* dimension-size-200 */
+
+  
+}
+
+.spectrum--large {
+  --spectrum-slider-tick-mark-height: 13px;
+  --spectrum-slider-tick-mark-width: 2px;
+  --spectrum-slider-tick-mark-border-radius: 2px;
+  --spectrum-slider-ramp-track-height: 20px;
+}
```
I _think_ we're going to need to pull these out of here, and place them into the Tokens package. I'll reach out to you about this on Slack.

bernhard-adobe replied:
thanks, it is worth knowing that the Design team has not yet decided on the tokens for the tick and ramp variant as of now. That's why they were coded outside the token package @pfulton

lunarfusion commented on components/slider/index.css:
```diff
@@ -10,63 +10,94 @@ OF ANY KIND, either express or implied. See the License for the specific languag
+
+/* missing tokens for variant with Tick, also need t-shirt sizing */
+.spectrum--medium {
+  --spectrum-slider-tick-mark-height: 10px;
```
Have we checked with design to see if the tick marks are being removed? I don't see them represented in the XD redline or at https://spectrum.adobe.com/page/slider/

If we are keeping them, there may be a way to calculate their height relative to the controls height.

lunarfusion commented on components/slider/index.css:
```diff
@@ -10,63 +10,94 @@ OF ANY KIND, either express or implied. See the License for the specific languag
+  /* --spectrum-global-dimension-size-25: */
+  --spectrum-slider-tick-mark-border-radius: 2px;
+  /* --spectrum-global-dimension-size-25 2px for Spectrum,   --spectrum-global-dimension-size-10 1px for Express */
+  --spectrum-slider-ramp-track-height: 16px;
```
`--spectrum-slider-ramp-track-height: 16px;`

I think this might need to just be a block size of 100%. After the t-shirt sizes are applied to the controls, it seems to work out right.

bernhard-adobe replied:
<img width="520" alt="Screen Shot 2023-01-24 at 17 21 30" src="https://user-images.githubusercontent.com/52184321/214451678-fed9447a-aeb0-40f0-9579-dd1f39cc1933.png">

I don't think that works. The control height is a different size and the ramp height. 
We should still keep the var `--spectrum-slider-ramp-track-height` and replace it with tokens.

lunarfusion commented on components/slider/index.css:
```diff
@@ -10,63 +10,94 @@ OF ANY KIND, either express or implied. See the License for the specific languag
+  /* --spectrum-global-dimension-size-125 */
+  --spectrum-slider-tick-mark-width: 2px;
+  /* --spectrum-global-dimension-size-25: */
+  --spectrum-slider-tick-mark-border-radius: 2px;
```
`--spectrum-slider-tick-mark-border-radius: 2px;`

Might this be redundant since it is the same as the width?

bernhard-adobe replied:
I think those should be still two different vars as we will replace those hard-coded values soon with tokens.

lunarfusion commented on components/slider/index.css:
```diff
@@ -10,63 +10,94 @@ OF ANY KIND, either express or implied. See the License for the specific languag
+.spectrum--medium {
+  --spectrum-slider-tick-mark-height: 10px;
+  /* --spectrum-global-dimension-size-125 */
+  --spectrum-slider-tick-mark-width: 2px;
```
This appears the same as **slider-track-thickness.** I think we could use the same token here.

bernhard-adobe replied:
I think we should keep it as an a var as we will replace those hard-coded values soon with tokens.

lunarfusion commented on components/slider/index.css:
```diff
@@ -89,40 +120,39 @@ governing permissions and limitations under the License.
 .spectrum-Slider-track,
 .spectrum-Slider-fill {
-  block-size: var(--spectrum-slider-track-height);
+  block-size: var(--mod-slider-track-thickness, var(--spectrum-slider-track-thickness));
```
I think this needs to be declared at the top of the file like

```
.spectrum-Slider {
  --spectrum-slider-track-fill-thickness: var(--spectrum-slider-track-thickness);
}
```

And then used like

```
.spectrum-Slider-track,
.spectrum-Slider-fill {
  block-size: var(--mod-slider-track-fill-thickness, var(--spectrum-slider-track-fill-thickness));
}
```

bernhard-adobe replied:
not sure actually. the var `mod-slider-track-thicknes` is coming directly from the core-tokens vars. 

``` 
"slider-track-thickness": {
    "sets": {
      "spectrum": {
        "value": "2px"
      },
      "express": {
        "value": "4px"
      }
    }
  },
````

@pfulton  is it ok to reference vars directly inside the Component from the variables.json file and adding the mods or do we need to assign it to another CSS variable first? I am fine either way, it's an easy fix.

bernhard-adobe replied:
moved to top.

lunarfusion commented on components/slider/index.css:
```diff
@@ -366,52 +388,55 @@ governing permissions and limitations under the License.
   position: relative;
 
-  inline-size: var(--spectrum-slider-tick-mark-width);
+  inline-size: var(--mod-slider-tick-mark-width, var(--spectrum-slider-tick-mark-width));
```
I think this can be the same token as the slider track thickness, like

```
.spectrum-Slider-tick {
  inline-size: var(--mod-slider-track-fill-thickness, var(--spectrum-slider-track-fill-thickness));
}
```

bernhard-adobe replied:
I think we should keep this a separate token as we are adding the tick marks to the core-tokens package soon.

lunarfusion commented on components/slider/index.css:
```diff
@@ -366,52 +388,55 @@ governing permissions and limitations under the License.
     content: "";
-    inline-size: var(--spectrum-slider-tick-mark-width);
-    block-size: var(--spectrum-slider-tick-mark-height);
+    inline-size: var(--mod-slider-tick-mark-width, var(--spectrum-slider-tick-mark-width));
```
Same as above, I think this can be  `inline-size: var(--mod-slider-track-fill-thickness, var(--spectrum-slider-track-fill-thickness));`

bernhard-adobe replied:
Same as above, I think we should keep this a separate token as we are adding the tick marks to the core-tokens package soon.

lunarfusion reviewed (changes requested):
I think that some refactoring might help simplify things and result in a more accurate overall height for the component.

bernhard-adobe commented on components/slider/index.css:
```diff
@@ -184,43 +215,29 @@ governing permissions and limitations under the License.
+        border-start-end-radius: var(--mod-slider-track-corner-radius, var(--spectrum-slider-track-corner-radius));
+        border-end-end-radius: var(--mod-slider-track-corner-radius, var(--spectrum-slider-track-corner-radius));
         border-start-start-radius: 0;
         border-end-start-radius: 0;
```
That is correct! Thank you, the radii still look correct without them. 
<img width="1614" alt="Screen Shot 2023-01-24 at 16 53 24" src="https://user-images.githubusercontent.com/52184321/214447047-a4dd7310-2cb8-40cc-8032-f79b515c62d7.png">

bernhard-adobe commented on components/slider/index.css:
```diff
@@ -1,5 +1,5 @@
 /*
-Copyright 2019 Adobe. All rights reserved.
+Copyright 2022 Adobe. All rights reserved.
```
how time flies.. yes

bernhard-adobe commented on components/slider/package.json:
```diff
@@ -17,7 +17,7 @@
   },
   "peerDependencies": {
-    "@spectrum-css/vars": "^8.0.0"
+    "@spectrum-css/tokens": "^5.0.0"
```
Tested, looks good in Slider. no changes

bernhard-adobe commented:
https://github.com/adobe/spectrum-css/pull/1547#discussion_r1085617662

@lunarfusion not sure why I was not able to comment directly on the t-shirt sizing of the slider-control heights.
I defined them, and now referenced them in `.spectrum-Slider-controls`. 

I think we still ned the 200px slider height defined as ` --spectrum-slider-height-size: var(--spectrum-component-height-200);` as we are calculating a couple of positions of that.  We could maybe have a different height for t-shirt sizes S,M,L, XL. 

I did remove the height and ramp-height vars and it leads to mis-calculations as the slider controls aka the handle are not related to the overall Slider height and Ramp height.

@pfulton  we should maybe still add the tokens:

````
.spectrum--medium {
  --spectrum-slider-tick-mark-height: 10px;
  /* --spectrum-global-dimension-size-125 */
  --spectrum-slider-tick-mark-width: 2px;
  /* --spectrum-global-dimension-size-25: */
  --spectrum-slider-tick-mark-border-radius: 2px;
  /* --spectrum-global-dimension-size-25 2px for Spectrum,   --spectrum-global-dimension-size-10 1px for Express */
  --spectrum-slider-ramp-track-height: 16px;
  /* dimension-size-200 */
}

.spectrum--large {
  --spectrum-slider-tick-mark-height: 13px;
  --spectrum-slider-tick-mark-width: 2px;
  --spectrum-slider-tick-mark-border-radius: 2px;
  --spectrum-slider-ramp-track-height: 20px;
}

bernhard-adobe commented on components/slider/index.css:
```diff
@@ -455,17 +459,250 @@ governing permissions and limitations under the License.
+    background: var(--mod-slider-ramp-handle-background-color, var(--spectrum-slider-ramp-handle-background-color));
+
+    @media (forced-colors: active) {
+      /* forced-color-adjust required to ensure the "circle" around the handle is transparent */
```
@lunarfusion I think I found one instance where the `forced-color-adjust` is required. 
In this case to show the "transparent border" around the Ramps's Handle.

**Without**
Emulated:
<img width="431" alt="Screen Shot 2023-01-25 at 16 24 02" src="https://user-images.githubusercontent.com/52184321/214716870-c96b5323-0e92-4d83-abd4-0a70ba21505c.png">
Windows 11 Edge on ARM:

<img width="505" alt="Screen Shot 2023-01-25 at 16 26 55" src="https://user-images.githubusercontent.com/52184321/214716893-fc2f13f8-97ad-4add-93eb-f7b9587f40f7.png">


**Width**
Emulated: 

<img width="381" alt="Screen Shot 2023-01-25 at 16 24 30" src="https://user-images.githubusercontent.com/52184321/214716959-81cf4f39-3eaf-4d74-a88d-75ee90505411.png">


Windows 11 Edge on ARM:
<img width="460" alt="Screen Shot 2023-01-25 at 16 27 20" src="https://user-images.githubusercontent.com/52184321/214717053-b8e3ae6e-1427-44b5-908f-4d9459a42b51.png">

lunarfusion reviewed (changes requested):
See what you think, just a couple things. Looking good!

bernhard-adobe commented on components/slider/index.css:
```diff
@@ -10,63 +10,77 @@ OF ANY KIND, either express or implied. See the License for the specific languag
+  --spectrum-slider-handle-border-width-down: var(--spectrum-slider-handle-border-width-down-small);
+}
+
+.spectrum-Slider--sizeM {
```
We could, usually not a requirement in CSS IMO to have a fallback but sure, great idea.
We should bring this up in your next CSS meeting (adding a note to my notes)

bernhard-adobe commented on components/slider/index.css:
```diff
@@ -75,8 +89,8 @@ governing permissions and limitations under the License.
   display: block;
-  min-block-size: var(--spectrum-slider-height);
-  min-inline-size: var(--spectrum-slider-min-width);
+  min-block-size: var(--mod-slider-height-size, var(--spectrum-slider-height-size));
```
oh that is an excellent point.
You can see this well comparing the previous size for Slider on https://opensource.adobe.com/spectrum-css/slider.html against the new height:

previous:
<img width="335" alt="Screen Shot 2023-01-26 at 13 59 53" src="https://user-images.githubusercontent.com/52184321/214949457-784e03b7-2601-4ea3-ab05-3d55f78436c7.png">
new:
<img width="360" alt="Screen Shot 2023-01-26 at 14 00 02" src="https://user-images.githubusercontent.com/52184321/214949498-7db022cb-f103-45be-98c7-90fbefeaaa71.png">

bernhard-adobe replied:
now matching height 32 
<img width="334" alt="Screen Shot 2023-01-26 at 22 15 25" src="https://user-images.githubusercontent.com/52184321/215014611-409fa2f9-ce8d-4e2c-a180-9de0552442a6.png">

lunarfusion replied:
Isn't the height of the controls supposed to be 16px at medium size + medium theme? I am seeing this in the tokens:

```
.spectrum-medium {
  --spectrum-slider-control-height-medium: 16px;
}
```

Screenshot of what I am seeing:
![Screenshot 2023-01-27 at 2 08 10 PM](https://user-images.githubusercontent.com/25614178/215187268-91204293-e8b7-484e-ae8e-306d65a61b4b.png)

bernhard-adobe replied:
16px medium and the other t-shirts sizes  is easy to set using the var `-spectrum-slider-control-height`. 
<img width="331" alt="Screen Shot 2023-01-30 at 14 26 46" src="https://user-images.githubusercontent.com/52184321/215605194-f39a953d-4920-44e7-ad0c-833fd072e45d.png">

bernhard-adobe commented on components/slider/index.css:
```diff
@@ -455,17 +459,250 @@ governing permissions and limitations under the License.
+  &::before {
+    background: var(--highcontrast-slider-track-color, var(--mod-slider-track-color, var(--spectrum-slider-track-color)));
+
+    @media (forced-colors: active) {
```
We would then re-create the entire nesting in that block with `::before` nesting. 
It felt to me harder to figure out the context of what is over-writing where as context is a bit lost when it is all in the same block.

@lunarfusion with placing them in a single one `@media`block we would have to also add in that block all of these over-writes? Thanks for the clarification!
```
.spectrum-Slider--ramp {
  .spectrum-Slider-handle {
     &::before {
        ...overwrites here...
```

bernhard-adobe replied:
moved in the bottom block with var-overwrites. Thanks for go with me through the pros and cons @lunarfusion !

bernhard-adobe commented:
@lunarfusion We are discussing in QOTD solutions on the slider-control-area size, the margins that come after the item and the margin between the label and the Slider that are impacted because of the height changes to 16px.
We should have an answer soon and I will also provide the update in our standup tomorrow.

https://adobedesign.slack.com/archives/GEN034MAS/p1675290685922999

bernhard-adobe commented:
> @lunarfusion We are discussing in QOTD solutions on the slider-control-area size, the margins that come after the item and the margin between the label and the Slider that are impacted because of the height changes to 16px. We should have an answer soon and I will also provide the update in our standup tomorrow.
> 
> https://adobedesign.slack.com/archives/GEN034MAS/p1675290685922999

We decided to update:
> to ensure that the slider control keeps its 32px click area height, we can’t define the control height to be smaller
32px hard-coding of the control area


>the slider control height (top artboard) will need to be updated from 16px (first image) to 32px (second image)

> the field label to control spacing (bottom artboard) will need to be updated from 0 (first image) to 8px (second image)

@lunarfusion Patrick suggested to hard-code the updates for now as we won't have an updated tokens package for now. 
I will make the updates today.

bernhard-adobe commented:
final values for label position to slider control height confirmed by PJ are:
small: −5 px / −6 px
medium: −8 px / −10 px
large: −11 px / −14 px
extra-large: −14 px / −17 px

pfulton commented on components/slider/index.css:
```diff
@@ -345,11 +356,22 @@ governing permissions and limitations under the License.
-  padding-block-start: var(--spectrum-fieldlabel-m-padding-top);
+  padding-block-start: var(--mod-component-top-to-text-75, var(--spectrum-component-top-to-text-75));
+  font-size: var(--mod-font-size-75, var(--spectrum-font-size-75));
+  margin-block-end: var(--mod-component-bottom-to-text-75, var(--spectrum-component-bottom-to-text-75));
+
+ /* February 2023: custom scaled tokens for label margin block end 
+  that will be replaced with an updated token for Slider component-bottom-to-text-75 */
+
+  margin-block-end: var(--mod-spectrum-slider-label-margin-block-end, var(--spectrum-slider-label-margin-block-end));
```
Looks like we probably want to either remove or comment out one of these two `margin-block-end` declarations. Due to the way that the cascade works, the browser's going to use the later one, but for authoring clarity, I think we need to either remove or comment out one of these.

bernhard-adobe commented:
@pfulton I added a calculation for the correct margin-block-end between the label and the slider controls.
The values from design are based on a margin from baseline to control.

![line-height-adjustment](https://user-images.githubusercontent.com/52184321/216706255-cc361ab5-25a9-49a8-ac00-5cc2d82a63c7.png)

Over-laying the results from the calc show that the label now is placed at the right spot and the slider controls are  set to 32px height.


<img width="1321" alt="Screen Shot 2023-02-03 at 12 19 11" src="https://user-images.githubusercontent.com/52184321/216705798-3a398720-37f4-48c4-9518-b33a148cf790.png">

I have removed the other margin-block as this should work until we can remove the hard-coded tokens in `custom-large.css` and `custom-medium.css`.

lunarfusion commented on components/slider/index.css:
```diff
@@ -366,52 +400,55 @@ governing permissions and limitations under the License.
-    1px
-  );
+  margin-inline: var(--mod-slider-track-margin-offset, var(--spectrum-slider-track-margin-offset)) ;
+  margin-block-start: calc(var(--spectrum-slider-height-size) / 2 - var(--spectrum-slider-tick-mark-height) / 2);
```
Line 414 and line 501 appear to be causing the ticks version to have extra space from the label to the controls. Visual:

![slider-ticks-label-distance](https://user-images.githubusercontent.com/25614178/216721421-0ad601f0-2fca-40da-9cc0-aea63d54b442.png)

lunarfusion commented on components/slider/index.css:
```diff
@@ -455,17 +495,239 @@ governing permissions and limitations under the License.
 
   position: absolute;
-  top: calc(var(--spectrum-slider-track-height) / 2 - 1px);
+  inset-block-start: calc(var(--mod-slider-control-height, var(--spectrum-slider-control-height)) / 2 - 1px);
```
Lines 414 and 501 are causing the tick variant to have extra space from label to controls. When I comment out lines 414 and 501, they move to where they are supposed to be, but the tick positions will need to be adjusted:

![Screenshot 2023-02-03 at 4 36 28 PM](https://user-images.githubusercontent.com/25614178/216724196-fc3e343c-0ba3-4918-ab15-07891f97ccdb.png)

And likewise the track and handle position will need to be adjusted as well:
![Screenshot 2023-02-03 at 4 37 15 PM](https://user-images.githubusercontent.com/25614178/216724329-3a989907-da3b-4725-b2a2-31723cd4d0db.png)

lunarfusion commented on components/slider/index.css:
```diff
@@ -455,17 +495,239 @@ governing permissions and limitations under the License.
+    .spectrum-Slider-labelContainer {
+      color: var(--highcontrast-slider-label-text-color-disabled, var(--mod-slider-label-text-color-disabled, var(--spectrum-slider-label-text-color-disabled)));
+    }
+
```
I am wondering if the tick labels should be added here like

```
    .spectrum-Slider-tickLabel {
      color: var(--highcontrast-slider-label-text-color-disabled, var(--mod-slider-label-text-color-disabled, var(--spectrum-slider-label-text-color-disabled)));
    }
```

To give the tick labels the same disabled color as the rest of the text. I've asked for feedback from design, we will see what they say.

lunarfusion replied:
Hey @bernhard-adobe , Design has confirmed that the tick labels should be the same color as the other disabled labels. This could be added either here or on line 522.

lunarfusion reviewed (changes requested):
Can we see if the tick variants can have the same label-to-control spacing as the others? Thanks!

bernhard-adobe commented:
thanks @lunarfusion for pointing out the ticks + marks in relation to label.
I set these with the `:has()` selector and they now match the new provided specs from design in regards the position from label to track.

(We don't have token specs yet for the tick mark variant and tick mark labels)

Here an overlay how they match over the new specs. 
<img width="1948" alt="Screen Shot 2023-02-03 at 20 01 27" src="https://user-images.githubusercontent.com/52184321/216746927-9d615a0b-dd53-4bf0-8a2d-49a57438cf4b.png">

lunarfusion reviewed (changes requested):
Everything looks really good, nicely lined up with consistent spacing from the label to the controls. This was a really tricky one, lots of moving parts!

Just one change since design responded about disabled tick labels.

pfulton commented:
Updated beta released: `@spectrum-css/slider@4.0.0-beta.2`

pfulton commented:
Updated beta released after resolving malformed CSS: `@spectrum-css/slider@4.0.0-beta.3`

pfulton commented:
One more beta bump: `@spectrum-css/slider@4.0.0-beta.4`

castastrophe commented on components/slider/package.json:
```diff
@@ -18,11 +18,13 @@
   },
   "peerDependencies": {
-    "@spectrum-css/vars": "^8.0.0"
+    "@spectrum-css/stepper": "^4.0.1",
+    "@spectrum-css/tokens": "^7.6.0"
```
```suggestion
    "@spectrum-css/stepper": "^4.0.0",
    "@spectrum-css/tokens": "^7.0.0"
```

As much as possible I'd like to keep peerDependencies at their lowest compatible release.

castastrophe commented on site/resources/js/enhancement.js:
```diff
@@ -558,7 +558,7 @@ function makeDoubleSlider(slider) {
 
     var x = Math.max(Math.min(e.x - sliderOffsetLeft, sliderOffsetWidth), 0);
-    var percent = (x / sliderOffsetWidth) * 100;
+    var percent = Math.round((x / sliderOffsetWidth) * 100);
```
Nice!  Excellent addition here.
